java/node/src/main/java/org/svarm/node/engine/impl/V1SingleEntryEngine.java

```diff
@@ -97,52 +100,17 @@ public String definitionName() {
    *
    * @param tenantTable table to read from.
    * @param entity      the entity id.
-   * @return a entry if found.
+   * @return an entry if found.
    */
   @Override
   public Optional<EntryInfo> read(final TenantTable tenantTable, final String entity) {
     LOGGER.trace("read({},{})", tenantTable, entity);
-    return sqlEngine.executePreparedTenant(tenantTable,
-        "select * from TENANT_DATA where ID = ?",
-        (ps) -> {
-          try {
-            long timestamp = 0;
-            int hash = 0;
-            ps.setString(1, entity);
-            try (final ResultSet rs = ps.executeQuery()) {
-              int rows = 0;
-              final ObjectNode node = jsonEngine.createObjectNode();
-              while (rs.next()) {
-                rows++;
-                final String col = rs.getString("C_COL");
-                final String type = rs.getString("C_DATA_TYPE");
-                final String data = rs.getString("C_DATA");
-                hash = rs.getInt("HASH");
-                timestamp = Math.max(timestamp, rs.getLong("TIMESTAMP"));
-                switch (type) {
-                  case INTEGER_TYPE -> node.put(col, Integer.valueOf(data));
-                  case STRING_TYPE -> node.put(col, data);
-                  default -> {
-                    LOGGER.error("Unknown: {}:{}:{}:{}", tenantTable, entity, col, type);
-                    throw new IllegalArgumentException("Unknown type: " + type + " for " + entity + ":" + tenantTable);
-                  }
-                }
-              }
-              if (rows > 0) {
-                return Optional.of(ImmutableEntryInfo.builder()
-                    .data(node)
-                    .timestamp(timestamp)
-                    .locationHash(hash)
-                    .id(entity)
-                    .build());
-              } else {
-                return Optional.empty();
-              }
-            }
-          } catch (SQLException e) {
-            throw new IllegalArgumentException("Unable to read data for: " + tenantTable + ":" + entity, e);
-          }
-        });
+
+    return Jdbi.create(dataSourceManager.getDataSource(tenantTable))
```

Comment:
I'm not clear if by using the static method Jdbi.create() you get all the build stuff you see in the DataSourceModule... the immutibles stuff should fail.

At least we should pull in that jdbi instance to this class, and use that. (Here and below)

Reply:
To add more context, we will want things like metrics... example call I want to enable. 
`
   jdbi.setSqlLogger(new InstrumentedSqlLogger(metricRegistry));
`
I'm guessing we'll need to cache them with their data sources? (Instead of caching the datasource, we cache the jdbi? As long as we can delete it that is)

Reply:
I can see that, but I think it makes more sense to add a `getJdbi` method to the DataSourceManager, allow us to cache it and instrument construction as necessary.  I think the Jdbi interface is actually quite cheap to construct since the datasource should manage caching/pooling connections. 

Probably a task for another issue though.

Reply:
We can add it in later, for sure. But note we are actually going to configure the JDBI instances... so not sure its as cheap as you think.